teamMaster.py

```diff
@@ -0,0 +1,264 @@
+    """
+    --date  optional, will extract teams found in boxscore for this date
+    """
+
+    parser = argparse.ArgumentParser('teamMaster command line arguments')
+    parser.add_argument(
+        '-d',
+        '--date',
+        help='Date of daily schedule for input - mm-dd-yyyy format',
+        dest='game_date',
+        type=str)
+
+    argue = parser.parse_args()
+    logger.info('Command arguments: ' + str(argue.game_date))
+
+    return argue
+
+
+def determine_filenames(gamedate=None):
+    """
+    :param game_date: date of the games in format "MM-DD-YYYY"
+    """
+
+    if gamedate is not None:
+        yyyy = gamedate[6:10]
+        mm = gamedate[0:2]
+        dd = gamedate[3:5]
+    else:
+        yyyy = datetime.date.today().strftime("%Y")
+        mm = datetime.date.today().strftime("%m")
+        dd = datetime.date.today().strftime("%d")
+
+    mmddyyyy = mm + '-' + dd + '-' + yyyy
+    schedule_input = DAILY_SCHEDULE.replace('YYYYMMDD', yyyy + mm + dd)
+    team_output = TEAM_MSTR_O.replace('YYYYMMDD', yyyy + mm + dd)
+
+    logger.info('dailySched dictionary location: ' + schedule_input)
+    logger.info('Updated team dictionary location: ' + team_output)
+
+    return (schedule_input, team_output, mmddyyyy)
+
+
+def load_daily_schedule(schedule_in):
+    """
+    :param schedule_in: filename for daily schedule file to load into memory
+    """
+
+    logger.info('Loading dailySchedule dictionary')
+
+    try:
+        with open(schedule_in) as schedulefile:
+            return json.load(schedulefile)
+    except Exception as e:
+        errmsg = 'Error loading dailySchedule dictionary. . .'
+        logger.critical(errmsg)
+        logger.exception(e)
+        raise LoadDictionaryError(errmsg)
+
+
+def reset_todays_games_in_master(masterdict):
+
+    keylist = list(masterdict)
+    for key in keylist:
+        keys = list(masterdict[key])
+        if 'today_1' in keys:
+            del masterdict[key]['today_1']
+        if 'today_2' in keys:
+            del masterdict[key]['today_2']
+
+    return masterdict
+
+
+def process_schedule(sched_dict, mstr_dict):
+    """
+    :param sched_dict: daily schedule file used to drive team extract process
+    """
+
+    mstr_dict = reset_todays_games_in_master(mstr_dict)
+
+    logger.info('Applying updates to team master dictionary')
+
+    # get dailySchedule key list and start with blank result team dictionary
+    keylist = list(sched_dict)
+    daily_team_dict = {}
+
+    # for each key, go to boxscore for that game and exctract all team data
+    for key in keylist:
+
+        boxscoreurl = BOXSCORE.replace('/_directory_/',
+                                       sched_dict[key]['directory'])
+        logger.info('Loading boxscore dictionary: ' + boxscoreurl)
+
+        try:
+            boxresp = requests.get(boxscoreurl)
+            boxscore_dict = json.loads(boxresp.text)
+            box_level_3 = boxscore_dict['data']['boxscore']
+        except Exception:
```

Comment:
I have seen the list in that horrible sea of documentation before. I just stuck with the broad exception in this case because the data is not mine to correct if there's an issue, and if there is I can't do anything about it. But the list of standard is duly noted and bookmarked in my growing library of useful Python links.